### PR TITLE
Add Promise/*/resolve-not-callable-close.js

### PR DESCRIPTION
**Pull request by marjakh**

### CODE REVIEW

**Comment by shvaikalesh** [comment]
Related: https://github.com/tc39/ecma262/pull/1912.

File: test/built-ins/Promise/any/resolve-not-callable-close.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Promise.resolve = "certainly not callable";
+
+Promise.any(iter).then(() => {
+  $DONE('The promise should be rejected, but was resolved');
+}, (reason) => {
+  assert(reason instanceof TypeError);
+}).then($DONE, $DONE);
+
+assert.sameValue(returnCount, 1);
```
Comment (rwaldron) [inline test/built-ins/Promise/any/resolve-not-callable-close.js:44]: Duplicate of: https://github.com/tc39/test262/tree/master/test/built-ins/Promise/any/invoke-resolve-get-error-close.js

File: test/built-ins/Promise/all/resolve-not-callable-close.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+Promise.resolve = "certainly not callable";
+
+Promise.all(iter).then(() => {
+  $DONE('The promise should be rejected, but was resolved');
+}, (reason) => {
+  assert(reason instanceof TypeError);
+}).then($DONE, $DONE);
+
+assert.sameValue(returnCount, 1);
```
Comment (rwaldron) [inline test/built-ins/Promise/all/resolve-not-callable-close.js:47]: Duplicate of: https://github.com/tc39/test262/blob/master/test/built-ins/Promise/all/invoke-resolve-get-error-close.js

File: test/built-ins/Promise/allSettled/resolve-not-callable-close.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+Promise.resolve = "certainly not callable";
+
+Promise.allSettled(iter).then(() => {
+  $DONE('The promise should be rejected, but was resolved');
+}, (reason) => {
+  assert(reason instanceof TypeError);
+}).then($DONE, $DONE);
+
+assert.sameValue(returnCount, 1);
```
Comment (rwaldron) [inline test/built-ins/Promise/allSettled/resolve-not-callable-close.js:44]: Duplicate of: https://github.com/tc39/test262/blob/master/test/built-ins/Promise/allSettled/invoke-resolve-get-error-close.js

File: test/built-ins/Promise/race/resolve-not-callable-close.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+Promise.resolve = "certainly not callable";
+
+Promise.race(iter).then(() => {
+  $DONE('The promise should be rejected, but was resolved');
+}, (reason) => {
+  assert(reason instanceof TypeError);
+}).then($DONE, $DONE);
+
+assert.sameValue(returnCount, 1);
```
Comment (rwaldron) [inline test/built-ins/Promise/race/resolve-not-callable-close.js:47]: Duplicate of: https://github.com/tc39/test262/blob/master/test/built-ins/Promise/race/invoke-resolve-get-error-close.js

File: test/built-ins/Promise/all/resolve-not-callable-close.js (hunk 2 above)
Comment (shvaikalesh) [reply test/built-ins/Promise/all/resolve-not-callable-close.js:47]: Existing test covers abrupt completion at step 5 of [PerformPromiseAll](https://tc39.es/ecma262/#sec-performpromiseall), while this PR adds coverage for step 6. Same with other methods.

Comment (marjakh) [reply test/built-ins/Promise/all/resolve-not-callable-close.js:47]: Exactly, like @shvaikalesh says. This is covering the case where "resolve" is not callable, while existing tests cover the case that Get(constructor, "resolve") throws.

As an additional data point, we had a situation in V8 where the existing test262 tests were passing but this test failed B-) so it's surely not a duplicate. For more info: https://bugs.chromium.org/p/v8/issues/detail?id=10452

File: test/built-ins/Promise/any/resolve-not-callable-close.js (hunk 1 above)
Comment (camillobruni) [reply test/built-ins/Promise/any/resolve-not-callable-close.js:44]: This is a different test, the Promise.resolve is not callable, while in the old test it just throws.

File: test/built-ins/Promise/allSettled/resolve-not-callable-close.js (hunk 3 above)
Comment (camillobruni) [reply test/built-ins/Promise/allSettled/resolve-not-callable-close.js:44]: This is a different test, the Promise.resolve is not callable, while in the old test it just throws.

File: test/built-ins/Promise/race/resolve-not-callable-close.js (hunk 4 above)
Comment (camillobruni) [reply test/built-ins/Promise/race/resolve-not-callable-close.js:47]: This is a different test, the Promise.resolve is not callable, while in the old test it just throws.

**Comment by syg** [comment]
@rwaldron could you take a look again? These are slightly different but not identical tests, so shouldn't be duplicates.

**Comment by rwaldron** [comment]
@syg sorry for the delay, I usually try to do Test262 triaging all at once each week.

**Pull request merged**